lib/chef/provider/package/debian.rb

```diff
@@ -0,0 +1,135 @@
+#
+# Author:: Kapil Chouhan (<kapil.chouhan@msystechnologies.com>)
+# Copyright:: Copyright 2008-2018, Chef Software Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "chef/provider/package"
+
+class Chef
+  class Provider
+    class Package
+      class Debian < Chef::Provider::Package
+        use_multipackage_api
+
+        action :install do
+          unless target_version_array.any?
+            logger.trace("#{new_resource} is already installed - nothing to do")
+            return
+          end
+
+          if new_resource.response_file
+            if preseed_file = get_preseed_file(package_names_for_targets, versions_for_targets)
+              converge_by("preseed package #{package_names_for_targets}") do
+                preseed_package(preseed_file)
+              end
+            end
+          end
+
+          converge_by(install_description) do
```

Comment:
Can we put this `converge_by` block under some conditional to maintain the idempotent behavior?